src/main/groovy/jenkins/automation/utils/ScmUtils.groovy

```diff
@@ -28,7 +28,8 @@ class ScmUtils {
                         branch("*/tags/$version")
                     }
                     relativeTargetDir(repo.sub_directory)
-                    shallowClone(repo.shallow?:false)
+                    //shallow clone is the default
+                    shallowClone(repo.shallow!=false?true:false)
```

Comment:
This is too much. The original code  shallowClone(repo.shallow?:false) should do the trick
What elvis does is check if expression is true and assign itself to a value, otherwise some other value. 
So if repo.shallow is true, shallowClone is true, otherwise false.
Groovy evaluates 0 and null as false, so we should be ok.